src/adapters/mc/asb/CMakeLists.txt

```diff
@@ -19,15 +19,26 @@ target_link_libraries(OsConfigResourceAsb
         asb)
 
 # SO binary name must match the universal class name in MOF with a "lib" prefix added
-set_target_properties(OsConfigResourceAsb PROPERTIES OUTPUT_NAME libOsConfigResource)
+set_target_properties(OsConfigResourceAsb PROPERTIES OUTPUT_NAME OsConfigResource)
 
-# Create the policy artifacts ZIP package for the Azure Security Baseline for Linux
+# The binary size limit for the SO binary, in bytes 
+set(MAX_FILE_SIZE 762880)
+file(WRITE "${PROJECT_BINARY_DIR}/check_filesize.cmake"
+    "file(READ filesize.txt actual_file_size)\n"
+    "if(actual_file_size GREATER ${MAX_FILE_SIZE})\n"
+        "message(FATAL_ERROR \"SO binary size exceeds the maximum allowed size of ${MAX_FILE_SIZE} bytes\")\n"
+    "endif()\n")
 
+# Create the policy artifacts ZIP package for the Azure Security Baseline for Linux
 add_custom_target(stage_create_asb_zip
     COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/Staging
     COMMAND ${CMAKE_COMMAND} -E copy_if_different "${CMAKE_CURRENT_SOURCE_DIR}/AzureLinuxBaseline.metaconfig.json" ${PROJECT_BINARY_DIR}/Staging/
     COMMAND ${CMAKE_COMMAND} -E copy_if_different "${CMAKE_CURRENT_SOURCE_DIR}/AzureLinuxBaseline.mof" ${PROJECT_BINARY_DIR}/Staging/
     COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:OsConfigResourceAsb> ${PROJECT_BINARY_DIR}/Staging/Modules/DscNativeResources/OsConfigResource/libOsConfigResource.so
+    COMMAND ${CMAKE_COMMAND} -E echo "Checking file size of $<TARGET_FILE:OsConfigResourceAsb>"
+    COMMAND stat --format=%s $<TARGET_FILE:OsConfigResourceAsb> > ${PROJECT_BINARY_DIR}/filesize.txt
```

Comment:
Can we change this to `filesize_asb.txt` to be consistent with ssh?

Reply:
Actually, think I will remove these from here and we will use a GitHub action in the future. It's not reliable, CI builds fails because of it as they have bigger sizes than the release binaries.

Reply:
Sounds good, i think thats a better idea